### ISSUE TITLE
GOPATH from environment broken with VSCode 1.22

### ISSUE THREAD

**Issue by markeissler** [Microsoft/vscode-go#1621]
As of __VSCode__ 1.22, GOPATH is no longer set correctly from the environment. Restoring VSCode 1.21.1 fixes the problem. I'm not sure if this is a VSCode bug or API change that needs to be reflected in vscode-go, therefore I'm posting the bug here.

I use the Go Environment Manager [GVM2](https://github.com/markeissler/gvm2) to maintain the value of `$GOPATH`. Basically, when you `cd` into a Go project directory, GVM2 will automatically update `$GOPATH` and related variables based on a configuration (`.go-version` and `.go-pkgset` files). What happens is no different than if a user set `$GOPATH` manually.

Up until __VSCode 1.21.1__ I could `cd` into a project directory, change the `$GOPATH`, launch VSCode with `code .`, and VSCode would properly load/parse the currently set `$GOPATH`. __As of VSCode 1.22__ (tested on 1.22.1, specifically), the behavior I'm seeing is the equivalent of VSCode first changing to a different directory (possibly the user home directory) and then reading `$GOPATH` before loading the project's contents. Of course, when VSCode changes to a different directory (possibly to launch a new shell and grab the resulting environment variables) GVM will update `$GOPATH` to something else, often to reflect the default system install of Go and its path information, which is 100% correct for a different directory (like the user's home directory) but 100% incorrect for the project in the current directory from which VSCode was launched.

What you're left with is imports that won't resolve because `$GOPATH` is incorrect.

__I installed VSCode 1.21.1 which simply loads all of the same extensions that I have for `1.22.1` and everything just works again.__

I can't see from the __VSCode 1.22__ release notes what may have changed to cause this. I thought I'd start here in case this is related to some sort of breaking API change that needs to be accounted for in __vscode-go__.

**Comment by ramya-rao-a** [comment]
Thanks for reporting the issue @markeissler. I do recall some conversations around VS Code changing directories when being started as you described. I'll try to get more information and get back to you. Until then, can you share which OS you are working on?

**Comment by ramya-rao-a** [comment]
@markeissler Without GVM, I cannot replicate this using your steps. I'll try installing gvm and get back here

**Comment by markeissler** [comment]
@ramya-rao-a This is strange. I'm not seeing the behavior now either. This is after seeing the problem for an entire day. I was able to switch back and forth between 1.22.1 (where the problem appeared) and 1.21.1 (where there was no problem). I did it repeatedly, switching back and forth. Very bizarre.

I'm going to close this issue and re-open if the problem appears again. Thanks for looking into this.
